src/Dialect/ONNX/Rewrite.td

```diff
@@ -117,6 +117,12 @@ def HaveSameLastDim: Constraint<
         "[$1.getType().cast<RankedTensorType>().getRank() - 1])">,
   "Two tensors have the same last dimension">;
 
+class HaveSameDim<int dim>: Constraint<
+  CPred<"onnx_mlir::hasShapeAndRank($0) && onnx_mlir::hasShapeAndRank($1) && "
+        "($0.getType().cast<RankedTensorType>().getShape()[" # dim # "] =="
+        "$1.getType().cast<RankedTensorType>().getShape()[" # dim # "])">,
```

Comment:
Do we need to add a check for dynamic dimensions, i.e. `getShape()[dim] != -1`? In case of dynamic dims, we don't know dim sizes are equal or not.

Reply:
Added